Update AOTInductor compile logic for CPU backend for Meta internal env

hl475 opened the pull request:
Reviewed By: muchulee8, chenyang78

Differential Revision: D49944410




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

chenyang78 commented on torch/_inductor/codecache.py:
```diff
@@ -1026,9 +1030,14 @@ def compile(
-            objcopy_command = build_paths.objcopy()
+            if not cuda and graph.aot_mode:  # Meta internal AOTInductor CPU
+                objcopy_command = build_paths.objcopy_fallback()
+                use_absolute_path = True
```
I think it's fine at the moment. I think we should refactor this file to have a better way to isolate meta-related stuff. Also, it seems to be hard to connect ```use_absolute_path``` with ```os.chmod(output_o, 0o644)``` below. I would wonder why we need to apply chmod to the output_o if we use absolute path? A better name or adding some comment would help. Thanks.

hl475 committed:
Update AOTInductor compile logic for CPU backend for Meta internal env (#110729)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/110729

Reviewed By: muchulee8, chenyang78

Differential Revision: D49944410
